File: backend/schemas/stock.py
from datetime import datetime
from decimal import Decimal

from pydantic import BaseModel


class StockResponse(BaseModel):
    """
    Complete stock information returned by FinSight.
    """

    id: int
    symbol: str
    company_name: str

    company_logo_url: str | None = None
    exchange: str | None = None
    sector: str | None = None
    industry: str | None = None
    currency: str | None = None

    latest_price: Decimal | None = None
    last_refreshed_at: datetime | None = None

    created_at: datetime
    updated_at: datetime

    class Config:
        from_attributes = True

File: frontend/src/features/market/types/stock.ts
export type StockQuote = {
  id: number;
  symbol: string;
  company_name: string;

  company_logo_url: string | null;
  exchange: string | null;
  industry: string | null;
  currency: string | null;

  latest_price: number | string | null;
  last_refreshed_at: string | null;

  created_at: string;
  updated_at: string;
};

export type StockSearchResult = {
  id?: number;
  symbol: string;
  display_symbol?: string | null;
  company_name: string;
  asset_type?: string | null;
  exchange?: string | null;
  sector?: string | null;
  industry?: string | null;
  currency?: string | null;
  latest_price?: number | string | null;
  company_logo_url?: string | null;
  last_refreshed_at?: string | null;
  is_enriched?: boolean;
};
